website/src/app/kb/deploy/dns/readme.mdx

```diff
@@ -0,0 +1,93 @@
+import PlanBadge from "@/components/PlanBadge";
+import Alert from "@/components/DocsAlert";
+import SupportOptions from "@/components/SupportOptions";
+import NextStep from "@/components/NextStep";
+
+<PlanBadge plans={["starter", "enterprise"]}>
+
+# Configure DNS
+
+</PlanBadge>
+
+Firezone includes a sophisticated DNS routing system available on all plans that
+provides Split DNS and fallback resolver configuration for each client
+application in your Firezone account. Read more below to understand how it works
+and how to configure it.
+
+## How DNS works in Firezone
+
+Each Firezone client embeds a lightweight DNS proxy used to route queries to an
+appropriate gateway for resolution.
+
+The proxy "listens on" IP addresses in the following ranges:
+
+- IPv4: `100.100.111.0/24`
+- IPv6: `fd00:2021:1111:8000::100:100:111:0/120`
+
+When a user signs in, the client configures the host operating system to use
+this proxy as the default resolver for all queries on the system.
+
+This is why you'll commonly see `100.100.111.1` or
+`fd00:2021:1111:8000::100:100:111:0` as the DNS server responding to DNS queries
+when the Firezone client is signed in.
+
+### Routing DNS queries
+
+When the proxy sees a query, it checks to see if it matches a resource the user
+has access to. If it does, the proxy then finds an appropriate gateway
+associated with the resource, forwards the query to be resolved on the gateway,
+then returns the result back to the client to be returned to the application
+making the query.
+
+DNS responses generated by the proxy have a TTL of **5 minutes** to ensure that
+clients will occasionally re-query the proxy to find the best gateway to serve a
```

Comment:
It will not trigger a gateway reconnection, DNS TTL has nothing to do with our traffic routing, it's just for cases when upstream DNS changes IPs (and we must actually read TTL and use it in the future, and set 3 sec TTL on clients for resource domains).

Reply:
Hm this raises an interesting point.

@conectado How does failover work for DNS resources? If I've created a DNS mapping to a gateway, and the gateway goes down, will the mapping automatically recover? Or will the application need to request a new IP for the resource?

Reply:
If the application needs to request a new IP for the resource, it will only do so after 5 minutes, correct? And many applications simply won't.

Refs #3369

Reply:
The TTL of the local response is set to 1second, every 5 minutes we refresh the dns mapping, but if the connection to the gateway stops the mapping is invalidated.

Most stateful connections will need to be re-established if the connection fails and will therefore re-resolve the address. Save for ping there aren't many applications that after connections stop and start again don't resolve the address.

The mapping can't be automatically recovered, though we could try to do a best effort, because the mapping is between IPs and not DNS names.

Reply:
> It will not trigger a gateway reconnection, DNS TTL has nothing to do with our traffic routing, it's just for cases when upstream DNS changes IPs (and we must actually read TTL and use it in the future, and set 3 sec TTL on clients for resource domains).

Yes, this, it's an open issue #3010 we should refresh the dns mappings every TTL seconds instead of 5 minutes but this is independent from the TTL we set for the record

Reply:
> Save for ping there aren't many applications that after connections stop and start again don't resolve the address.

I think this was my question: it's the DNS query itself that triggers the resource_intent correct? So as long as as the connection is healthy, a new gateway will never be picked.

If the connection to the gateway times out, the application will need to requery DNS again, which typically (we're assuming) will happen when it tries to re-establish its connectivity.

Do I have that right?

Reply:
> I think this was my question: it's the DNS query itself that triggers the resource_intent correct? So as long as as the connection is healthy, a new gateway will never be picked.

> If the connection to the gateway times out, the application will need to requery DNS again, which typically (we're assuming) will happen when it tries to re-establish its connectivity.

> Do I have that right?

Precisely!